File: using_python_as_a_calculater/ex.java
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;

public class Ex {

    public static void main(String[] args) {
        boolean theWorldIsFlat = true;
        if (theWorldIsFlat) {
            System.out.println("Be careful not to fall off!");
        }

        // Strings in JAVA
        String word = "Python";
        System.out.println(word.charAt(0));
        System.out.println(word.charAt(word.length() - 1));

        System.out.println(word.substring(0, 2));
        System.out.println(word.substring(2, 5));
        System.out.println(word.substring(0, 5));
        System.out.println(word.substring(word.length() - 2, word.length())); 
        System.out.println("word length: " + word.length());
        // ERROR: String index out of range:-4
        // System.out.println(word.substring(word.length() - 2, 0)); 
       
        // strings is a object  
        
        // Lists in JAVA
        List<Integer> squares = Arrays.asList(1, 4, 9, 16, 25);
        System.out.println(squares);
        System.out.println(squares.get(0));
        System.out.println(squares.get(squares.size() - 1));
        System.out.println(Arrays.asList(Arrays.copyOfRange(squares.toArray(), squares.size() - 3, squares.size())));

        List<Integer> squaresCopy = Arrays.asList((Integer[])Arrays.copyOfRange(squares.toArray(), 0, squares.size()));
        System.out.println(squaresCopy);

        List<Integer> squaresConcat = new ArrayList<>(Arrays.asList((Integer[])squares.toArray())); // Arrays.ArrayList的add居然还有坑
        squaresConcat.addAll(Arrays.asList(36, 49, 64, 81, 100));
        System.out.println(squaresConcat);

        List<Integer> cubes = new ArrayList<>(Arrays.asList(1, 8, 27, 65, 125));
        System.out.println(cubes);
        cubes.set(3, 64);
        System.out.println(cubes);
        cubes.add(216);
        cubes.add((int)Math.pow(7, 3));
        System.out.println(cubes);

        List<Character> letters = new ArrayList<>(Arrays.asList('a', 'b', 'c', 'd', 'e', 'f', 'g'));
        System.out.println(letters);

        // 官方的标准库不支持区域替换
         
        // Fibonacci 数列
        int a = 0, b = 1; 
        while (b < 1000) {
            System.out.print(b + ",");
            int c = a + b;
            a = b;
            b = c;
        }

        System.out.println();
    }
}

File: using_python_as_a_calculater/ex.py






the_world_is_flat = True
if the_world_is_flat:
    print("Be careful not to fall off!")



# strings in Python
word = "Python"
print(word[0])
print(word[-1])

print(word[:2])
print(word[2:5])
print(word[:5])
print(word[-2:])
print("word length:", len(word))
# print nothing
# print(word[-2:0])

# strings is immutable

# Lists in Python
squares = [1, 4, 9, 16, 25]
print(squares)
print(squares[0])
print(squares[-1])
print(squares[-3:]) # slicing returns a new list

squaresCopy = squares[:] # copy list
print(squaresCopy)

squaresConcat = squares + [36, 49, 64, 81, 100]
print(squaresConcat)

cubes = [1, 8, 27, 65, 125]
print(cubes)
cubes[3] = 64
print(cubes)
cubes.append(216)
cubes.append(7 ** 3)
print(cubes)

letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g']
print(letters)
letters[2:5] = ['C', 'D', 'E']
print(letters)
letters[2:5] = []
print(letters)
letters[:] = []
print(letters)

# Fibonacci 数列
a, b = 0, 1
while b < 1000:
    print(b, end=",")
    a, b = b, a + b

